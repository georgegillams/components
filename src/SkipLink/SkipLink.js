/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016-2020 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import PropTypes from 'prop-types';
import React from 'react';

import { cssModules } from '../helpers/cssModules';

import STYLES from './skip-link.scss';

const getClassName = cssModules(STYLES);

const SkipLink = props => {
  const { label, href, className, ...rest } = props;

  return (
    <a href={href} className={getClassName('skip-link', className)} {...rest}>
      {label}
    </a>
  );
};

SkipLink.propTypes = {
  label: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
  className: PropTypes.string,
};

SkipLink.defaultProps = {
  className: null,
};

export default SkipLink;
