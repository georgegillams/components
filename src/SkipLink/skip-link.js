import PropTypes from 'prop-types';
import React from 'react';
import { Link, StyledSkipLink, Wrapper } from './skip-link.styles';

const SkipLink = (props) => {
  const { label, href, className, ...rest } = props;

  return (
    <Wrapper {...rest}>
      <Link href={href}>{label}</Link>
    </Wrapper>
  );
};

SkipLink.propTypes = {
  label: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
};

SkipLink.defaultProps = {};

export default SkipLink;
