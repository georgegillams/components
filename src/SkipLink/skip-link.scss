/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016-2020 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

@import '../Tokens/common';

.skip-link {
  display: inline-block;
  padding: $spacing-sm * 2;
  transition: none;
  border-radius: $bpk-border-radius-sm;
  background-color: $bpk-color-panjin;
  color: white;
  font-weight: bold;
  text-decoration: none;
  opacity: 1;
  overflow: hidden;

  &--hidden {
    transition: opacity $duration-sm ease-in-out;
    opacity: 0;
    pointer-events: none;
  }
}
