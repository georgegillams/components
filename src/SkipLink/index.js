import SkipLink from './SkipLink';

export default SkipLink;
export { SkipLink };
