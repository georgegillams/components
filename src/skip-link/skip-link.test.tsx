import React from 'react';
import renderer from 'react-test-renderer';

import SkipLink from './index';

describe('SkipLink', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<SkipLink href="#main" label="Skip to main content" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should support arbitrary props', () => {
    const tree = renderer
      .create(
        // @ts-ignore
        <SkipLink href="#main" label="Skip to main content" testID="123" />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
