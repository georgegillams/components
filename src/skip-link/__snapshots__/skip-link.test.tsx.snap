// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkipLink should render correctly 1`] = `
.c1 {
  display: inline-block;
  border-radius: 0.25rem;
  font-weight: bold;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  color: #9932cc;
  fill: #9932cc;
}

.c1:focus-visible {
  position: relative;
  outline: none !important;
}

.c1:focus-visible::after {
  position: absolute;
  top: -0.35rem;
  right: -0.35rem;
  bottom: -0.35rem;
  left: -0.35rem;
  content: '';
  display: block;
  border: solid white 0.2rem;
  border-radius: 0.5rem;
  pointer-events: none;
}

.c1:hover {
  color: #44aeff;
  fill: #44aeff;
}

.c3 {
  margin: 0;
}

.c0 {
  background-color: #d1435b;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all 0.4s ease-in-out;
  transition: all 0.4s ease-in-out;
  -webkit-transition-delay: 0.2s;
  transition-delay: 0.2s;
  opacity: 0;
  height: 0;
  padding: 0;
}

.c0:focus-within,
.c0:focus {
  padding: calc(2 * 0.375rem);
  -webkit-transition: none;
  transition: none;
  opacity: 1;
  -webkit-transition-delay: 0;
  transition-delay: 0;
  height: 3rem;
}

.c2 {
  color: white;
  font-weight: bold;
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media (prefers-color-scheme:dark) {
  .c1:focus-visible::after {
    border-color: white;
  }
}

@media (prefers-color-scheme:dark) {
  .c1 {
    color: #da70d6;
    fill: #da70d6;
  }

  .c1:hover {
    color: #44aeff;
    fill: #44aeff;
  }
}

<div
  className="c0"
>
  <a
    className="c1 c2"
    href="#main"
  >
    <span
      className="c3"
      size="inherit"
    >
      Skip to main content
    </span>
  </a>
</div>
`;

exports[`SkipLink should support arbitrary props 1`] = `
.c1 {
  display: inline-block;
  border-radius: 0.25rem;
  font-weight: bold;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  color: #9932cc;
  fill: #9932cc;
}

.c1:focus-visible {
  position: relative;
  outline: none !important;
}

.c1:focus-visible::after {
  position: absolute;
  top: -0.35rem;
  right: -0.35rem;
  bottom: -0.35rem;
  left: -0.35rem;
  content: '';
  display: block;
  border: solid white 0.2rem;
  border-radius: 0.5rem;
  pointer-events: none;
}

.c1:hover {
  color: #44aeff;
  fill: #44aeff;
}

.c3 {
  margin: 0;
}

.c0 {
  background-color: #d1435b;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all 0.4s ease-in-out;
  transition: all 0.4s ease-in-out;
  -webkit-transition-delay: 0.2s;
  transition-delay: 0.2s;
  opacity: 0;
  height: 0;
  padding: 0;
}

.c0:focus-within,
.c0:focus {
  padding: calc(2 * 0.375rem);
  -webkit-transition: none;
  transition: none;
  opacity: 1;
  -webkit-transition-delay: 0;
  transition-delay: 0;
  height: 3rem;
}

.c2 {
  color: white;
  font-weight: bold;
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media (prefers-color-scheme:dark) {
  .c1:focus-visible::after {
    border-color: white;
  }
}

@media (prefers-color-scheme:dark) {
  .c1 {
    color: #da70d6;
    fill: #da70d6;
  }

  .c1:hover {
    color: #44aeff;
    fill: #44aeff;
  }
}

<div
  className="c0"
>
  <a
    className="c1 c2"
    href="#main"
  >
    <span
      className="c3"
      size="inherit"
    >
      Skip to main content
    </span>
  </a>
</div>
`;
